File: eoq-discount/src/main/java/local/example/outcome/OutcomeResource.java
package local.example.outcome;

import local.example.outcome.model.Eoq;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Set;

@Path("/outcome")
public class OutcomeResource {

    private final Set<Eoq> eoqs = Collections.newSetFromMap(
            Collections.synchronizedMap(new LinkedHashMap<>())
    );

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response read() {
        if (this.eoqs.isEmpty())
            return Response.ok().build();
        return Response.ok(this.eoqs).build();
    }
}

File: mle/src/main/kotlin/local/example/outcome/OutcomeResource.kt
package local.example.outcome

import local.example.outcome.model.Outcome

import java.util.*
import javax.ws.rs.*

import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/outcome")
class OutcomeResource {

    private val outcome = Collections.newSetFromMap(
        Collections.synchronizedMap(
            LinkedHashMap<Outcome, Boolean>()
        )
    )

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun read(): Response? {
        return if (outcome.isEmpty()) Response.ok().build() else Response.ok(outcome).build()
    }

    @POST
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    fun compute(eoq: Outcome): Set<Outcome?>? {
        eoq.setOutcome()
        outcome.add(eoq)
        return outcome
    }
}